# HG changeset patch
# User Rik <rik@octave.org>
# Date 1467850547 25200
#      Wed Jul 06 17:15:47 2016 -0700
# Node ID 55e2979e1fd180d889730a6ed65c1b00c4a66749
# Parent  90eca9a9c47aa617309f3d5e4afca642a1347700
legend.m: Don't use 'legend' tag for any non-axes, non-legend objects.

* legend.m: Use "deletelegend" tag for hidden text object used to
invoke destructor on legend when axes is destroyed.

diff --git a/scripts/plot/appearance/legend.m b/scripts/plot/appearance/legend.m
--- a/scripts/plot/appearance/legend.m
+++ b/scripts/plot/appearance/legend.m
@@ -912,20 +912,20 @@ function [hleg, hleg_obj, hplot, labels]
               xk = 0;
               yk += 1;
             endif
           endif
         endfor
 
         ## Add an invisible text object to original axis
         ## that when it is destroyed will remove the legend
-        props = {"parent", ca(1), "tag", "legend", ...
+        props = {"parent", ca(1), "tag", "deletelegend", ...
                  "handlevisibility", "off", "visible", "off", ...
                  "xliminclude", "off", "yliminclude", "off"};
-        t1 = findall (ca(1), "tag", "legend", "type", "text");
+        t1 = findall (ca(1), "tag", "deletelegend", "type", "text");
         if (isempty (t1))
           t1 = text (0, 0, "", props{:});
           set (t1, "deletefcn", {@deletelegend1, hlegend});
         endif
         if (isprop (hlegend, "unmodified_axes_position"))
           set (hlegend, "unmodified_axes_position",
                          unmodified_axes_position,
                         "unmodified_axes_outerposition",
